lab_1/main.py

```diff
@@ -4,80 +4,107 @@
 """
 
 
-def tokenize(text: str) -> list or None:
-    """
-    Splits a text into tokens, converts the tokens into lowercase,
-    removes punctuation and other symbols from words
-    :param text: a text
-    :return: a list of lower-cased tokens without punctuation
-    """
-    pass
+def tokenize(text: str):
+    if isinstance(text, str):
+        text = text.lower()
+        text = text.replace("'", ' ')
+        n_text = ''
+        for i in text:
+            if i.isalpha() or i == ' ' or i == '\n':
+                n_text += i
+            else:
+                text.replace(i, ' ')
+        t_tokens = n_text.split()
+        return t_tokens
+    return None
```

Comment:
There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.